kolibri/core/assets/src/content-renderer.vue

```diff
@@ -54,49 +79,96 @@
+        // If the Vue component is attached to the DOM, and it is unrendered, then render now!
         if (this.ready && !this.rendered) {
           this.renderContent();
         }
       },
+      /**
+       * Method that renders the dynamically retrieved component into the wrapper component.
+       */
       renderContent() {
-        if (this.currentViewClass !== null) {
+        // Only render if we have a component type to render and the content is available.
+        if (this.currentViewClass !== null & this.available & !this.rendered) {
+          // We are rendering, so don't let setRendererComponent call this again.
           this.rendered = true;
+          // Start building up an object of all props that the content renderers might get passed.
           const propsData = {};
-          const enumerables = Object.keys(this.contentData);
-          const properties = Object.getOwnPropertyNames(this.contentData).filter(
+          // List enumerable properties of the props object.
+          const enumerables = Object.keys(this.$options.props);
+          // Only get non-inherited properties of the props object.
+          const properties = Object.getOwnPropertyNames(this.$options.props).filter(
+            // Only use non-enumerable, non-inherited properties of the props object.
             (name) => enumerables.indexOf(name) > -1
           );
           for (const key in properties) {
-            if (key !== 'extra_fields') {
-              propsData[key] = this.contentData[key];
+            // Loop through all the properties, see if one of them is extraFields.
+            if (key !== 'extraFields') {
+              // If it isn't just put it directly into the data.
+              propsData[key] = this[key];
             } else {
-              Object.assign(propsData, JSON.parse(this.contentData[key]));
+              // If it is, then parse it as JSON and assign each key to the propsData.
+              Object.assign(propsData, JSON.parse(this[key]));
             }
           }
+          // Add a defaultFile to the propsData, which is the first file in the availableFiles.
           propsData.defaultFile = this.availableFiles[0];
+          // Create an options object for the soon to be instantiated renderer component.
           const options = {
+            // Set the parent so that it is in the Vue family.
             parent: this,
+            // Let it mount on the DOM in the container div set up in the template.
             el: this.$els.container,
+            // Pass in the propsData!
             propsData,
           };
+          // Add the specified options for the Vue component that we received from the plugin
+          // into the options object.
           Object.assign(options, this.currentViewClass);
+          // Instantiate the Vue instance directly using the Kolibri Vue constructor.
           this.contentView = new this.Kolibri.lib.vue(options); // eslint-disable-line new-cap
```

Comment:
Maybe we should just switch over to literate code style!

Shame that never took off (CoffeeScript had a nice implementation of it).